OpenRA.Mods.Common/Traits/World/ColorPickerManager.cs

```diff
@@ -0,0 +1,189 @@
+#region Copyright & License Information
+/*
+ * Copyright 2007-2021 The OpenRA Developers (see AUTHORS)
+ * This file is part of OpenRA, which is free software. It is made
+ * available to you under the terms of the GNU General Public License
+ * as published by the Free Software Foundation, either version 3 of
+ * the License, or (at your option) any later version. For more
+ * information, see COPYING.
+ */
+#endregion
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using OpenRA.Graphics;
+using OpenRA.Primitives;
+using OpenRA.Support;
+using OpenRA.Traits;
+
+namespace OpenRA.Mods.Common.Traits
+{
+	[Desc("Configuration options for the lobby player color picker. Attach this to the world actor.")]
+	public class ColorPickerManagerInfo : TraitInfo<ColorPickerManager>, IRulesetLoaded
+	{
+		[Desc("Minimum and maximum saturation levels that are valid for use.")]
+		public readonly float[] HsvSaturationRange = { 0.3f, 0.95f };
+
+		[Desc("HSV value component for player colors.")]
+		public readonly float V = 0.95f;
+
+		[Desc("Perceptual color threshold for determining whether two colors are too similar.")]
+		public readonly float SimilarityThreshold = 0.314f;
+
+		[Desc("List of hue components for the preset colors in the palette tab. Each entry must have a corresponding PresetSaturations definition.")]
+		public readonly float[] PresetHues = { };
+
+		[Desc("List of saturation components for the preset colors in the palette tab. Each entry must have a corresponding PresetHues definition.")]
+		public readonly float[] PresetSaturations = { };
+
+		[PaletteReference]
+		public readonly string PaletteName = "colorpicker";
```

Comment:
Mods might want to support more than one colorpicker palette with different `RemapIndices` for e.g. `FactionPreviewActors` or the asset browser. This would be a good time to introduce an `IProvidesColorPickerPaletteInfo` interface to decouple the palette side from the UI side.